### ISSUE TITLE
PLM Review

### ISSUE THREAD

**Issue by JamesStrabala** [JamesStrabala/Capstone-Resorces#20]
### Title
A Comprehensive Review of Protein Language Models
### URL
https://doi.org/10.48550/arXiv.2502.06881
### Summary 
Prot2Vec is an early method of embedding proteins, based on Word2Vec. ProSST has a structure quantization module that uses attention to create structure tokens.
Encoder-only (like BERT) PLMs (like the ESM family) are primarily used for processing for later tasks. Multiple of these models innovate on its masking capability. Decoder-only models are mostly used for generation. Here is a list:
* ProGen: Trained with unsupervised self-regression.
* ProGen 2: Has 6.4B parameters.
* RITA: Uses UniProt100's data, large-scale model training.
* ProtGPT2: Generates in seconds.
* PoET: I have this one covered.
* IgLM: Synthetic antibody libraries.
* ZymCTRL: Zero-shot enzyme making.
* ProLLama: Two stage training, can do multiple tasks as a ProLLM.
Encoder-Decoder models are generally used for translation.
* pAbT5: Uses T5 to generate complementary light or heavy chains. (Antibody components)
* xTrimoPGLM: I've covered this one.
* Ankh: A fine-tuning strategy to tune for a specific protein family.
* PreTriBERT: Uses 3D models to generate proteins for a function.
There are two main techniques for encoding position:
Absolute encoding:
Includes sequential information in the tokens. It's simple, but it creates difficulty when used on sequences of varying lengths. Transformers can over rely on this information so they underperform if elements are rearranged. Rotary Position Encoding (Rope) uses rotation matrices to encode position. This removes the weakness to varying lengths and can allow for dependencies between distant tokens.
Relative encoding:
Uses offsets between K and Q values to identify distances. Rising in popularity, but there are cases where absolute encoding is better.
PLMs scale up better than LLMs.  OpenAI proposed and validated the concept of Scaling laws to explain improvement vs size relationships.
Pre-training databases include:
* UniProt: Public and well-annotated. Has multiple subsets for various tasks.
* Pfam: Groups proteins into families with 2 alignments and  a Hidden Markov Model. Used for function and structure research.
* BFD: Integrates several public databases for billions of sequences.
### Key Points 
* List of models.
* Two schools of encoding.
* List of databses.
### Citation
@misc{wang2025comprehensivereviewproteinlanguage,
      title={A Comprehensive Review of Protein Language Models}, 
      author={Lei Wang and Xudong Li and Han Zhang and Jinyi Wang and Dingkang Jiang and Zhidong Xue and Yan Wang},
      year={2025},
      eprint={2502.06881},
      archivePrefix={arXiv},
      primaryClass={q-bio.BM},
      url={https://arxiv.org/abs/2502.06881}, 
}
### Repo link
https://github.com/ISYSLAB-HUST/Protein-Language-Models
